examples/controls_example/public/react_controls/control_group/serialization_utils.ts

```diff
@@ -0,0 +1,99 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0 and the Server Side Public License, v 1; you may not use this file except
+ * in compliance with, at your election, the Elastic License 2.0 or the Server
+ * Side Public License, v 1.
+ */
+
+import { Reference } from '@kbn/content-management-utils';
+import { DEFAULT_CONTROL_GROW, DEFAULT_CONTROL_WIDTH } from '@kbn/controls-plugin/common';
+import { SerializedPanelState } from '@kbn/presentation-containers';
+import { omit } from 'lodash';
+import { DefaultControlApi, DefaultControlState } from '../types';
+import { ControlGroupRuntimeState, ControlGroupSerializedState, ControlPanelState } from './types';
+
+export const deserializeControlGroup = <
+  ChildControlState extends ControlPanelState = ControlPanelState
+>(
+  state: SerializedPanelState<ControlGroupSerializedState>
+): ControlGroupRuntimeState<ChildControlState> => {
+  const panels = JSON.parse(state.rawState.panelsJSON);
+  const ignoreParentSettings = JSON.parse(state.rawState.ignoreParentSettingsJSON);
+
+  /** Inject data view references into each individual control */
+  const references = state.references ?? [];
+  references.forEach((reference) => {
+    const referenceName = reference.name;
+    const panelId = referenceName.substring('controlGroup_'.length, referenceName.lastIndexOf(':'));
+    if (panels[panelId]) {
+      panels[panelId].dataViewId = reference.id;
+    }
```

Comment:
cc @ThomThomson

Reply:
I imagine it would be at control group deserialization time - although, that would require grabbing each control's factory, which might not be ideal? I will leave it as-is for now I think, since it should work for **all** current controls with references... We can address this once we have controls that have a reference that **isn't** a data view :+1: